synapse/storage/databases/main/room.py

```diff
@@ -1533,68 +1432,31 @@ async def set_room_is_public_appservice(
                 list.
         """
 
-        def set_room_is_public_appservice_txn(txn, next_id):
-            if is_public:
-                try:
-                    self.db_pool.simple_insert_txn(
-                        txn,
-                        table="appservice_room_list",
-                        values={
-                            "appservice_id": appservice_id,
-                            "network_id": network_id,
-                            "room_id": room_id,
-                        },
-                    )
-                except self.database_engine.module.IntegrityError:
-                    # We've already inserted, nothing to do.
-                    return
-            else:
-                self.db_pool.simple_delete_txn(
-                    txn,
+        if is_public:
+            try:
+                await self.db_pool.simple_insert(
                     table="appservice_room_list",
-                    keyvalues={
+                    values={
                         "appservice_id": appservice_id,
                         "network_id": network_id,
                         "room_id": room_id,
                     },
+                    desc="set_room_is_public_appservice_true",
                 )
-
-            entries = self.db_pool.simple_select_list_txn(
-                txn,
-                table="public_room_list_stream",
+            except self.database_engine.module.IntegrityError:
```

Comment:
This works, but I'd rather we avoided it as integrity errors cause postgres to log errors to its logs, which is a bit annoying. Instead can you use `simple_upsert` with `values={}` please? That will do a `INSERT ... ON CONFLICT DO NOTHING`, which is what we probably want here.